experimental/traffic-portal/src/app/api/topology.service.ts

```diff
@@ -0,0 +1,185 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { HttpClient } from "@angular/common/http";
+import { Injectable } from "@angular/core";
+import type {
+	RequestTopology,
+	ResponseTopology,
+	ResponseTopologyNode,
+} from "trafficops-types";
+
+import { APIService } from "./base-api.service";
+
+/**
+ * TopTreeNode is used to represent a topology in a format usable as a material
+ * nested tree data source.
+ */
+export interface TopTreeNode {
+	name: string;
+	cachegroup: string;
+	children: Array<TopTreeNode>;
+	parents: Array<this>;
+}
+
+/**
+ * TopologyService exposes API functionality relating to Topologies.
+ */
+@Injectable()
+export class TopologyService extends APIService {
+
+	constructor(http: HttpClient) {
+		super(http);
+	}
+
+	/**
+	 * Gets one or all Topologies from Traffic Ops
+	 *
+	 * @param name The name of a single Topology to be returned.
+	 * @returns An Array of Topologies
+	 * whether `name` was	passed.
```

Comment:
You have a stray tab character in the middle of this sentence.

Reply:
Whitespace fixed in 4b9352e801